homeassistant/components/hydrawise/__init__.py

```diff
@@ -40,16 +40,15 @@
     "UNIT_OF_MEASURE_INDEX",
 ]
 DEVICE_MAP = {
-    "auto_watering": ["Automatic Watering", "mdi:autorenew", "", ""],
+    "auto_watering": ["Automatic Watering", "mdi:autorenew", "switch", ""],
```

Comment:
Entities with device class set should not set a custom icon. Use `None` in those cases.

No unit of measurement should be represented by `None`.

Reply:
Please import the device class constants from the respective entity integration module adn use here.

https://github.com/home-assistant/core/blob/7cc31022091ef77618f53d05163a758dbae04439/homeassistant/components/switch/__init__.py#L39